generator/internal/descriptor_utils.cc

```diff
@@ -629,53 +658,16 @@ std::string FormatApiMethodSignatureParameters(
         input_type->FindFieldByName(parameter);
     google::protobuf::SourceLocation loc;
     parameter_descriptor->GetSourceLocation(&loc);
-    auto comment = absl::StrReplaceAll(
-        loc.leading_comments,
-        {
-            {kDialogflowCXEnvironmentIdProto1, kDialogflowCXEnvironmentIdCpp1},
-            {kDialogflowCXEnvironmentIdProto2, kDialogflowCXEnvironmentIdCpp2},
-            {kDialogflowCXSessionIdProto, kDialogflowCXSessionIdCpp},
-            {kDialogflowCXTransitionRouteGroupIdProto,
-             kDialogflowCXTransitionRouteGroupIdCpp},
-            {kDialogflowCXEntityTypeIdProto, kDialogflowCXEntityTypeIdCpp},
-            {kDialogflowESSessionIdProto, kDialogflowESSessionIdCpp},
-            {kDialogflowESContextIdProto, kDialogflowESContextIdCpp},
-            {kDialogflowESSessionEntityTypeDisplayNameProto,
-             kDialogflowESSessionEntityTypeDisplayNameCpp},
-            {kArtifactRegistryRepositoryNameProto,
-             kArtifactRegistryRepositoryNameCpp},
-        });
-    comment = absl::StrReplaceAll(
-        EscapePrinterDelimiter(ChompByValue(comment)),
+    auto comment = EscapePrinterDelimiter(ChompByValue(loc.leading_comments));
+    for (auto& sub : substitutions) {
```

Comment:
Is there a performance cost to splitting this into N calls?

I think there would be N memory accesses per byte of the comment. But I dont think a comment length will exceed the L1 cache. So performance should be unaffected. Does that sound right?

Reply:
I haven't checked the `StrReplaceAll()` implementation, but I'm sure there is some extra cost, yes.  But then again, we probably gain more back by doing the substitutions in place, rather than returning copies.  In the end, though, I'm reckon none of that matters in the big picture.